lib/fact_changes.rb

```diff
@@ -535,38 +522,31 @@ def _fact_operations(action_type, step, facts)
     operations = facts.map do |fact|
       modified_assets.push(fact.object_asset) if listening_to_predicate?(fact.predicate)
       Operation.new(:action_type => action_type, :step => step,
-        :asset=> fact.asset, :predicate => fact.predicate, :object => fact.object, object_asset: fact.object_asset)
+                    :asset => fact.asset, :predicate => fact.predicate, :object => fact.object, object_asset: fact.object_asset)
     end
 
     modified_assets.flatten.compact.uniq.each(&:touch)
     operations
   end
 
-  def all_values_are_new_records(hash)
-    hash.values.all? do |value|
-      (value.respond_to?(:new_record?) && value.new_record?)
-    end
+  def any_values_are_new_records(hash)
```

Comment:
This was being used to determine if the fact was likely new. This would be the case if *any* of its associated values were new, not just all of them. (Plus in most cases we have non-record attributes anyway, where this would never be true) This was resulting in unecessary trips out to the database.